### PR TITLE
Removed flux scaling from upwind discretization

### LINKED ISSUE #563: Upwind discretization should not scale with flux
The single-point upwind discretization, `pp.Upwind`, should be modified to no longer scale with the flux. Specifically, the discretization matrices specified by keywords `'Upwind.upwind_matrix_key` and `Upwind.outflow_neumann_matrix_key` should contain values `{-1, 0, 1}`, so that the flux scaling can be factored in in other places.

### PR DESCRIPTION
**Pull request by keileg**
Main changes:
* The implicit scaling with the advective flux field in the upwind discretization is removed. 
* The scaling with the flux field is kept in the method `assemble_matrix_rhs()` and related methods. This was necessary to keep the expected functionality when using these methods (and thereby also the `Assembler`). This solution is a bit messy, but I see no other approach that does not put an undue burden on the user. 
* Expanded the tests somewhat.

Resolves #563.

### CODE REVIEW

**Comment by IvarStefansson** [comment]
> 
> 
> @IvarStefansson: While we're at it, I find the keyword for the right hand side discretization in `Upwind`, 'rhs' non-intuitive. Shall we change it, and if so, do you have suggestions?

Yes, something along the lines of "boundary_transport". I see mass matrix has bound_mass, fv_elliptic uses bound_flux.

**Comment by keileg** [comment]
> > @IvarStefansson: While we're at it, I find the keyword for the right hand side discretization in `Upwind`, 'rhs' non-intuitive. Shall we change it, and if so, do you have suggestions?
> 
> Yes, something along the lines of "boundary_transport". I see mass matrix has bound_mass, fv_elliptic uses bound_flux.

bound_transport it is then.

File: src/porepy/numerics/fv/upwind.py [hunk 1]
```diff
@@ -314,20 +347,24 @@ def discretize(self, g: pp.Grid, data: Dict, d_name: str = "darcy_flux") -> None
...
 
         # Use a simple cell to face map here; this will pick up cells that are both
         # upstream and downstream to Neumann faces, however, the latter will have
         # their flux filtered away.
         cell_2_face = np.abs(g.cell_faces)
 
         # Add minus sign to be consistent with other boundary terms
-        matrix_dictionary[self.outflow_neumann_matrix_key] = -sps.kron(
-            neumann_flux_mat * cell_2_face, sps.eye(num_components)
+        matrix_dictionary[self.outflow_neumann_matrix_key] = sps.kron(
```
[the remark this answers is not in the document]
Comment (keileg) [reply src/porepy/numerics/fv/upwind.py:366]: Having looked at this again, I think you are right: If a face is (known to be) outflow, it should be set to Dirichlet, and everything will be fine. Hence, I am tempted to remove all of this. Do you agree @IvarStefansson?

What happens in situation where the flow direction is not clear, e.g. for complex fields or in the case of advection-diffusion problems is another matter, but I think we can disregard that here.

[the remark this answers is not in the document]
Comment (IvarStefansson) [reply src/porepy/numerics/fv/upwind.py:366]: Yes, I agree

[the remark this answers is not in the document]
Comment (keileg) [reply src/porepy/numerics/fv/upwind.py]: I have deleted in last push.

**Review by IvarStefansson — APPROVED** [review]

**Pull request merged**